server/py/services/api/api/endpoints/model_endpoints.py

```diff
@@ -268,76 +262,71 @@ async def list_model_endpoints(
     )
 
     endpoints.endpoints = allowed_endpoints
+
     return endpoints
 
 
 async def _verify_model_endpoint_read_permission(
-    *, project: str, endpoint_id: str, auth_info: schemas.AuthInfo
+    *, project: str, name: str, auth_info: schemas.AuthInfo
 ) -> None:
     await (
         framework.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
             schemas.AuthorizationResourceTypes.model_endpoint,
             project_name=project,
-            resource_name=endpoint_id,
+            resource_name=name,
             action=schemas.AuthorizationAction.read,
             auth_info=auth_info,
         )
     )
 
 
 @router.get(
-    "/{endpoint_id}",
+    "/{name}",
+    status_code=HTTPStatus.OK.value,
     response_model=schemas.ModelEndpoint,
 )
 async def get_model_endpoint(
+    name: str,
     project: ProjectAnnotation,
-    endpoint_id: EndpointIDAnnotation,
-    start: str = Query(default="now-1h"),
-    end: str = Query(default="now"),
-    metrics: list[str] = Query([], alias="metric"),
+    function_name: Optional[str] = Query(default=None),
+    endpoint_id: Optional[EndpointIDAnnotation] = None,
+    tsdb_metrics: bool = Query(default=True),
```

Comment:
No need for Query for these simple defaults:
https://fastapi.tiangolo.com/tutorial/query-params/
```suggestion
    function_name: Optional[str] = None,
    endpoint_id: Optional[EndpointIDAnnotation] = None,
    tsdb_metrics: bool = True,
```

Reply:
There are still many API endpoints in this module with unnecessary `Query(...)` that can be removed.
You also `import typing`, but there is a `from typing import ...` in the module - please use the existing latter.